scripts/process-data/fitbitecg.R

```diff
@@ -0,0 +1,212 @@
+                      concept, 
+                      startdate, 
+                      enddate) %>% 
+      dplyr::reframe(numrecords = dplyr::n()) %>% 
+      dplyr::ungroup() %>% 
+      tidyr::pivot_longer(cols = c(numrecords), 
+                          names_to = "stat", 
+                          values_to = "value") %>% 
+      dplyr::mutate(concept = paste0("mhp:summary:weekly:", stat, ":", concept)) %>% 
+      dplyr::select(participantidentifier, startdate, enddate, concept, value) %>% 
+      dplyr::distinct()
+  }
+  
+  result <- 
+    dplyr::bind_rows(summarize_alltime(df), 
+                     summarize_weekly(df)) %>% 
+    dplyr::distinct()
+  
+  return(result)
+}
+
+library(dplyr)
+
+dataset <- "fitbitecg"
+
+cat(glue::glue("Transforming data for {dataset}"),"\n")
+
+# Get variables for this dataset
+vars <- 
+  selected_vars %>% 
+  filter(grepl(paste0(dataset, "$"), Export, ignore.case = TRUE, perl = TRUE)) %>% 
+  pull(Variable)
+
+# Load the desired subset of this dataset in memory and do some feature engineering for derived variables
+df <- 
+  arrow::open_dataset(file.path(downloadLocation, glue::glue("dataset_{dataset}"))) %>% 
+  select(all_of(c(vars))) %>% 
+  filter(ResultClassification %in% c("Normal Sinus Rhythm", "Atrial Fibrillation")) %>% 
+  rename(StartDate = StartTime) %>% 
+  mutate(EndDate = base_s3_uri %>% stringr::str_extract("[0-9]{4}-[0-9]{2}-[0-9]{2}")) %>%
+  collect()
+
+colnames(df) <- tolower(colnames(df))
+
+# Create lists for ID variables and i2b2 concept variables
+excluded_concepts <- c("participantidentifier", "startdate", "enddate", "starttime")
+
+approved_concepts_summarized <- 
+  setdiff(
+    tolower(selected_vars$Variable[selected_vars$Export==dataset]),
+    excluded_concepts
+  )
+
+# Pivot data frame from long to wide
+df_melted_filtered <- 
+  df %>% 
+  mutate("SinusRhythm" = if_else(resultclassification == "Normal Sinus Rhythm", 1, NA),
+         "AtrialFibrillation" = if_else(resultclassification == "Atrial Fibrillation", 1, NA)) %>% 
+  select(-c(resultclassification)) %>% 
+  recoverSummarizeR::melt_df(excluded_concepts = excluded_concepts) %>% 
+  select(if("participantidentifier" %in% colnames(.)) "participantidentifier",
+         dplyr::matches("(?<!_)date(?!_)", perl = T),
+         if("concept" %in% colnames(.)) "concept",
+         if("value" %in% colnames(.)) "value") %>% 
+  tidyr::drop_na("value") %>% 
+  mutate(value = as.numeric(value))
+cat("recoverSummarizeR::melt_df() completed.\n")
+
+# Generate i2b2 summaries
+df_summarized <- 
+  df_melted_filtered %>% 
+  rename(startdate = dplyr::any_of(c("date", "datetime"))) %>% 
+  mutate(enddate = if (!("enddate" %in% names(.))) NA else enddate) %>% 
+  select(all_of(c("participantidentifier", "startdate", "enddate", "concept", "value"))) %>% 
+  ecg_stat_summarize() %>% 
+  mutate(value = as.numeric(value)) %>% 
+  distinct()
+cat("recoverSummarizeR::stat_summarize() completed.\n")
```

Comment:
This is not the function you're calling. Is this output meaningful for processing logs?